### PR TITLE
Adding infra support for ec2, docker, virtualbox

### PR DESCRIPTION
**Pull request by gboddin**
- Support is done trough ec-europa/infra
- Support single instance for now
- Added default options in build.properties.dist
- Now building ec2 ami for cloudformation/codedeploy
- Now building docker images
- Now building virtualbox OVA

### CODE REVIEW

**Comment by pfrenssen** [comment]
When I check out this branch then I get an error about a missing file when I do `./bin/phing`:

```
$ git checkout demo
$ composer install
$ ./bin/phing
BUILD FAILED
/home/pieter/v/subsite-starterkit/build.xml:37:55: Unable to find build file: /home/pieter/v/subsite-starterkit/./vendor/ec-europa/infra/infra.xml
Total time: 0.0940 seconds
```

It looks like the `composer.lock` file is outdated.

**Comment by pfrenssen** [comment]
OK `composer.lock` is not part of the repository, I have updated it myself now.

The help texts are not replacing the build properties, the variable names are echoed in the console:

```
$ ./bin/phing

Main targets:
-------------------------------------------------------------------------------
8<--- snip
 docker-create-instance    Create a base docker image to run ${infra.type} applications
 docker-install-app        Create a ${application.name} instance from ${infra.type} image
 docker-run-app            Run an instance of ${application.name}
 docker-update-app         Update ${application.name} instance
```

Also it is not clear how to proceed now. In which order do I need to run these commands? Which build properties do I need to set?

Can you add a new section to the docs?

**Comment by gboddin** [comment]
Yes, of course :) ... Damn variables not set in the help section ... I'll make it generic then

File: build.properties.dist [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
+application.name = my-project
 
 # The make file containing the dependencies of the subsite.
 subsite.make = ${subsite.resources.dir}/site.make
 
 # Modules to enable after installation. Separate multiple modules with spaces.
 subsite.install.modules = myproject_core
 
+# The site type ( Infrastructure stack )
+infra.type = subsite
```
Comment (pfrenssen) [inline build.properties.dist]: Shall we add a documentation page on how to execute this?

**Comment by gboddin** [comment]
Closing, moving support to ec aws account + starting final centos infra in infra branch

**Pull request closed without merging**